checks/frozen.go

```diff
@@ -0,0 +1,170 @@
+	Private      bool              `json:"private"`
+	Bin          map[string]string `json:"bin"`
+	Dependencies map[string]string `json:"dependencies"`
+}
+
+// CheckFrozen is the exported name for Frozen-Check check.
+const CheckFrozen = "Frozen"
+
+type frozenPinningResult struct {
+	module pinnedResult
+	frozen pinnedResult
+}
+
+const (
+	module pinnedResult = iota
+	binary
+)
+
+//nolint:gochecknoinits
+func init() {
+	registerCheck(CheckFrozen, Frozen)
+}
+
+func addFrozenPinnedResult(w *frozenPinningResult, frozen, m bool) {
+	w.frozen = notPinned
+
+	if frozen {
+		w.frozen = pinned
+	}
+
+	if m {
+		w.module = module
+	} else {
+		w.module = binary
+	}
+
+}
+
+func isPackageJson(filename string) (bool, error) {
+	return strings.HasPrefix(strings.ToLower(filename), "package.json"), nil
+}
+
+func testIsFrozen(pathfn string, content []byte, dl checker.DetailLogger) (int, error) {
+	var r frozenPinningResult
+	_, err := validateFrozenIsPinned(pathfn, content, dl, &r)
+	return createReturnForIsFrozen(r, dl, err)
+}
+
+func createReturnForIsFrozen(r frozenPinningResult, dl checker.DetailLogger, err error) (int, error) {
+	return createReturnValuesForIsFrozen(r, "no frozen dependencies foound", dl, err)
+}
+
+func createReturnValuesForIsFrozen(r frozenPinningResult, infoMsg string, dl checker.DetailLogger, err error) (int, error) {
+	if err != nil {
+		return checker.InconclusiveResultScore, err
+	}
+
+	switch {
+	case (r.module == module) && (r.frozen == pinned):
+		return checker.MinResultScore, nil
+	case (r.module == module) && (r.frozen == notPinned):
+		return checker.MaxResultConfidence, nil
+	case (r.module == binary) && (r.frozen == pinned):
+		return checker.MinResultScore, nil
+	case (r.module == binary) && (r.frozen == notPinned):
+		return checker.MaxResultConfidence, nil
+	default:
+		return checker.MinResultScore, nil
+	}
+}
+
+func isFrozenDependencies(c *checker.CheckRequest) (int, error) {
+	var r frozenPinningResult
+	err := CheckFilesContent("package.json", false, c, validateFrozenIsPinned, &r)
+	return createReturnForIsFrozen(r, c.Dlogger, err)
+}
+
+func validateFrozenIsPinned(pathfn string, content []byte,
+	dl checker.DetailLogger, data FileCbData) (bool, error) {
+
+	pdata, ok := data.(*frozenPinningResult)
+	if !ok {
+		// panic if it is not correct type
+		panic("type need to be of frozenPinningResult")
+	}
+
+	var pjson = PackageJSON{}
+	err := json.Unmarshal(content, &pjson)
+
+	if err != nil {
+		return false, sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("%v: %v", errInternalInvalidYamlFile, err))
+	}
+
+	// use regex to check no dependencies are pinned
+	// ^([0-9]+)\.([0-9]+)\.([0-9]+)?$
+	regex := regexp.MustCompile(`^([0-9]+)\.([0-9]+)\.([0-9]+)?$`)
+	frozen := false
+
+	for _, v := range pjson.Dependencies {
+		// check if the dependency match
+		if regex.Match([]byte(v)) {
+			frozen = true
+			break
+		}
+	}
+
+	// check whether there is a bin information
```

Comment:
nit: comments start with capital letters and end with ".". One of the linters may complain.
Same comment for other comments.